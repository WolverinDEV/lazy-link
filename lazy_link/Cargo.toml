[package]
name = "lazy_link"
license-file.workspace = true
authors.workspace = true
repository.workspace = true
edition.workspace = true
version.workspace = true

[dependencies]
lazy_link_derive = { path = "../lazy_link_derive", version = "*" }

[dev-dependencies]
trybuild = "1.0.99"
