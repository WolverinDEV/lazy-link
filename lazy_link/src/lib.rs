#![no_std]

mod cache;
pub use cache::{
    Cache,
    NoCache,
    StaticAtomicCache,
    StaticCache,
};
pub use lazy_link_derive::lazy_link;
