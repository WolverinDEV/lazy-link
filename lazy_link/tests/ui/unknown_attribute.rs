use lazy_link::lazy_link;

#[lazy_link(resolver = "my_resolver", unknown = "12")]
extern "C" {
    fn method_01();
}

fn main() {}
