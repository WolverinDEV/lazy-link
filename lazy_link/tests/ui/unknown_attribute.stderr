error: unknown attribute
 --> tests/ui/unknown_attribute.rs:3:39
  |
3 | #[lazy_link(resolver = "my_resolver", unknown = "12")]
  |                                       ^^^^^^^
